### DISCUSSION TITLE
using tar_map and tar_plan

### DISCUSSION THREAD

**Discussion by mjrolland** [ropensci/targets#918] [Help] [answered]
Dear all,

I am currently working on a statistical analysis where I create many targets. I define them using `tarchetypes::tar_plan`. I am currently applying several times the same series of functions on subsets of the same input dataset. To do that I copy/paste the series of functions for each subset and define all the targets that way. It looks something like this.

```r
tar_plan(
  data = read_data("path"),
  hist_a = hist_fun(data, "a"),
  model_a = model_fun(data, "a"),
  hist_b = hist_fun(data, "b"),
  model_b = model_fun(data, "b"),
  hist_c = hist_fun(data, "c"),
  model_c = model_fun(data, "c")
)
```

From what I understand `tar_map` and static branching would save me having to copy/paste the same lines each time. However I am not able to use `tar_map` and `tar_plan` together, and I found no example doing so. All the examples do not use `tar_plan` but a list of `tar_targets`. 

So I am wondering: 
* is static branching the answer to my problem? 
* If yes, is it possible to use static branching within a `tar_plan`? 
* If yes, where can I find an example of that?

Thank you in advance for your help!

Matthieu

**Comment by wlandau** [comment, marked as answer]
`tar_plan()` can define all or part of a target list, so you can combine multiple `tar_plan()` calls together in the same target list. Here is what your example would look like with `tar_map()` and `tar_plan()`:

```r
# _targets.R file:
library(targets)
library(tarchetypes)
list(
  tar_plan(data = read_data("path")),
  tar_map(
    tar_plan(
      hist = hist_fun(data, arg),
      model = model_fun(data, arg)
    ),
    values = list(arg = c("a", "b", "c"))
  )
)
```

**Reply by mjrolland** [reply]
Thank you!
